pipeline/src/test_util.rs

```diff
@@ -71,13 +73,50 @@ pub fn gen_estark_proof(file_name: &str, inputs: Vec<GoldilocksField>) {
 pub fn gen_halo2_proof(file_name: &str, inputs: Vec<Bn254Field>) {
     let file_name = format!("{}/../test_data/{file_name}", env!("CARGO_MANIFEST_DIR"));
     let tmp_dir = mktemp::Temp::new_dir().unwrap();
-    Pipeline::default()
+    let mut pipeline = Pipeline::default()
         .with_tmp_output(&tmp_dir)
         .from_file(PathBuf::from(file_name))
         .with_prover_inputs(inputs)
-        .with_backend(powdr_backend::BackendType::Halo2)
-        .proof()
-        .unwrap();
+        .with_backend(powdr_backend::BackendType::Halo2);
```

Comment:
This is to test that the proof is valid in a separate step. Before we tested that the proof is generated, and setup and vkey were created during proof time. Now we test 3 paths:

- same as before, collect the proof
- verify the proof using the setup and vkey that were generated separately for the same circuit
- same as first, but using the generated setup and vkey, instead of generating them on the fly